diff --git a/Src/Modules/subreap.c b/Src/Modules/subreap.c
new file mode 100644
index 000000000..710461ce3
--- /dev/null
+++ b/Src/Modules/subreap.c
@@ -0,0 +1,68 @@
+#include<sys/prctl.h>
+#include "subreap.mdh"
+#include "subreap.pro"
+
+/**/
+static int
+bin_subreap(char *nam, char **args, Options ops, UNUSED(int func))
+{
+    int reaping = !OPT_ISSET(ops, 'u');
+    int result = prctl(PR_SET_CHILD_SUBREAPER, reaping);
+    return result;
+}
+
+/*
+ * boot_ is executed when the module is loaded.
+ */
+
+static struct builtin bintab[] = {
+    BUILTIN("subreap", 0, bin_subreap, 0, -1, 0, "u", NULL),
+};
+
+static struct features module_features = {
+    bintab, sizeof(bintab)/sizeof(*bintab),
+    0
+};
+
+/**/
+int
+setup_(UNUSED(Module m))
+{
+    return 0;
+}
+
+/**/
+int
+features_(Module m, char ***features)
+{
+    *features = featuresarray(m, &module_features);
+    return 0;
+}
+
+/**/
+int
+enables_(Module m, int **enables)
+{
+    return handlefeatures(m, &module_features, enables);
+}
+
+/**/
+int
+boot_(Module m)
+{
+    return 0;
+}
+
+/**/
+int
+cleanup_(Module m)
+{
+    return setfeatureenables(m, &module_features, NULL);
+}
+
+/**/
+int
+finish_(UNUSED(Module m))
+{
+    return 0;
+}
diff --git a/Src/Modules/subreap.mdd b/Src/Modules/subreap.mdd
new file mode 100644
index 000000000..fc721b1f8
--- /dev/null
+++ b/Src/Modules/subreap.mdd
@@ -0,0 +1,7 @@
+name=lilydjwg/subreap
+link=dynamic
+load=no
+
+autofeatures="b:subreap"
+
+objects="subreap.o"
